torchrl/modules/tensordict_module/rnn.py

```diff
@@ -399,3 +400,379 @@ def _lstm(
                 1,
             )
         return tuple(out)
+
+
+class GRUModule(ModuleBase):
+    """An embedder for an GRU module.
+
+    This class adds the following functionality to :class:`torch.nn.GRU`:
+
+    - Compatibility with TensorDict: the hidden states are reshaped to match
+      the tensordict batch size.
+    - Optional multi-step execution: with torch.nn, one has to choose between
+      :class:`torch.nn.GRUCell` and :class:`torch.nn.GRU`, the former being
+      compatible with single step inputs and the latter being compatible with
+      multi-step. This class enables both usages.
+
+
+    After construction, the module is *not* set in temporal mode, ie. it will
+    expect single steps inputs.
+
+    If in temporal mode, it is expected that the last dimension of the tensordict
+    marks the number of steps. There is no constrain on the dimensionality of the
+    tensordict (except that it must be greater than one for temporal inputs).
```

Comment:
so, just to clarify, if I have 
- an input tensor of shape [*B,T,*N,F]
- in a tensordict with shape [*B,T]

my *N dimensions are treated as batch? 
Is this usecase supported?

Reply:
Yes, see the `test_single_step_vs_multi` test

Reply:
in that test the *B is part of the tensordict.

Here I am asking about the *N 
like the agent dimension in MARL for example, after the T, but before the F